integrations/db-mongodb/README.md

```diff
@@ -0,0 +1,101 @@
+---
+title: 'MongoDB Database Integration'
+excerpt: 'The MongoDB Jovo integration allows you to store user specific data and more in a MongoDB database.'
+---
+
+# MongoDB Database Integration
+
+This [database integration](https://www.jovo.tech/docs/databases) allows you to store user specific data in a MongoDB table.
+
+## Introduction
+
+[MongoDB](https://www.mongodb.com/) is ... Shared Atlas clusters. Easy config.
+
+## Installation
+
+You can install the plugin like this:
+
+```sh
+$ npm install @jovotech/db-mongodb
+```
+
+Add it as plugin to any stage you like, e.g. `app.prod.ts`:
+
+```typescript
+import { MongoDb } from '@jovotech/db-mongodb';
+
+// ...
+
+app.configure({
+  plugins: [
+    MongoDb.instance({
```

Comment:
I would prefer `new MongoDb({ /* config */ })` here because it's our convention for Jovo plugins.

Could you tell us a bit more why you chose that approach?

Reply:
Sure. I did it that way because according to MongoDB best practices:

```
Define the client to the MongoDB server outside the [AWS Lambda handler function](http://docs.aws.amazon.com/lambda/latest/dg/nodejs-prog-model-handler.html).

Don't define a new MongoClient object each time you invoke your function. Doing so causes the driver to create a new database connection with each function call. This can be expensive and can result in your application exceeding database connection limits. As an alternative, do the following:

Create the MongoClient object once.
Store the object so your function can reuse the MongoClient across function invocations.
```
It's a way to keep the same connection pool across different invocations. AWS Lambda may reuse the same connection too if the container is not killed in successive invocations.

Any operation on the DB would use the same connection. For example:
1. When Jovo saves data in the lifecycle through assignments in jovo.$user.data.
2. When the user needs to query another MongoDB DB, another collection, or any database data they would not have to create a new instance (and another connection). It would be reused.


It's the only way I thought. Other ways I came up with would require the user to add more configuration, and generally scramble the code to achieve the same thing.

Reply:
By the way, The package is published for testing: https://www.npmjs.com/package/@jorgelg/mongodb

Reply:
Thank you. 

Or maybe we could offer a method like `this.$db.getInstance()` or `this.$mongoDb.getInstance()`?

Reply:
Yes, `this.$db.getInstance()` would be great.

I understand that the object would be created with 'new' like any other plugin, and from the accessor method the instance would be retrieved, returning something like `this.plugins.mongoDb`.

Actually though, I'm thinking that stages having two databases (rare, but could happen) would pose a problem calling `this.$db` because it wouldn't know which one to return.

It may be better to use `this.$mongoDb`

Reply:
I'm taking a look now. Could you please tell me how to get the MongoDB instance from App inside Jovo.ts?

Reply:
Another aproach would be:

1. use `this.mongoDb()`, because `this.$db` is more complicated for me, more checks would have to be done and 
and I can't think of an elegant solution.
2. a new interface in `DbPlugin` that looks something like this:
```
export interface DbInstance {
  getClient(): unknown;
  getJovoDb(): unknown;
  getJovoUsers(): unknown;
}
```
3. An abstract method also in `DbPlugin` class: `abstract getInstance(): DbInstance;`

Do you prefer something like this?

Reply:
Hi, @aswetlow 

Please, could you provide some advice in order to be able to get the DB client from the plugin in runtime?
I'm very new to TypeScript, and I found many difficulties on trying to do so.

The solution I raised would require DynamoDb class to implement the methods to return the client, I didn't see any problem there. But with FileDb a null would have to be returned, I think.

The most difficult part for me is changing the framework code to test it in my project. I tried for hours yesterday, but couldn't make it.

Thank you for your fantastic work, as always.

Reply:
Hi Jan, I think we can resolve this as we continued with another aproach down in the comments.